lightning/src/ln/channelmanager.rs

```diff
@@ -656,25 +656,74 @@ pub struct ChannelDetails {
 	pub counterparty_features: InitFeatures,
 	/// The value, in satoshis, of this channel as appears in the funding output
 	pub channel_value_satoshis: u64,
+	/// The value, in satoshis, which must always be held in the channel for us. This value ensures
+	/// that if we broadcast a revoked state, our counterparty can punish us by claiming at least
+	/// this value on chain.
+	///
+	/// This value is not included in [`outbound_capacity_msat`] as it can never be spent.
```

Comment:
I feel this *should* be included in `outbound_capacity_msat` as otherwise we are redefining "capacity". Or at least we need a term for capacity minus reserve (spending capacity?).

Reply:
I'm happy to rename it to spending capacity, but its used by the router to decide if it can fit an htlc into this channel so it should consider all the various limits, IMO.

Reply:
What about just `outbound_spendable_msat` and `inbound_spendable_msat`? I agree capacity is strange.

Reply:
> outbound_spendable_msat and inbound_spendable_msat

Fine with this solution, though arguably that would ideally take into account the fee spike buffer and such?

Reply:
Yea, we could try to include more things, but ultimately any use of this as an "exact" value is race-y - a send/forward could happen between when its read and when we go to send, and I feel like we'll screw it up pretty easily.

Reply:
Isn't this mixing up what the users is looking for (channel value = inbound + outbound capacity) with what we want to supply to the router interface (routable value)? I feel these are separate concepts and our `ChannelDetails` abstraction is being reused where maybe it shouldn't be.

Reply:
> Isn't this mixing up what the users is looking for (channel value = inbound + outbound capacity) with what we want to supply to the router interface (routable value)?

If that's what users are looking for, we've never provided that. These values have always not included pending HTLCs, meaning you can't meaningfully add them together to get anything. Still, I think what we expose makes sense - users want to know how much money they can currently send over a channel (modulo race conditions), that's a really useful number. It just so happens to also be useful for the router. We could debate whether the inbound_capacity value is useful, but the outbound one I'd argue is definitely right.

Reply:
Ah, I see. I was thinking more in terms of useful information to show an end-user (not necessarily the LDK developer) as we are doing in the sample node, which could be _what is my channel balance?_

I guess it is more of a question of what makes a good router interface -- which I'm happy to mull over -- and should it be part of a general `ChannelDetails` interface. Seems like we were using it for both but it wasn't satisfactory for either use case.

Reply:
> Ah, I see. I was thinking more in terms of useful information to show an end-user (not necessarily the LDK developer) as we are doing in the sample node, which could be what is my channel balance?

I'd still argue this is a reasonable thing, though maybe the *name* could be changed? As an end user I would very much like to know how much money I can send over a channel before I hit a limit. Phrasing the question as "how much of the channel is my balance?" seems vague to me - I'm not really sure that we can reasonably answer that question.

> I guess it is more of a question of what makes a good router interface -- which I'm happy to mull over -- and should it be part of a general ChannelDetails interface. Seems like we were using it for both but it wasn't satisfactory for either use case.

I think I disagree - the details of available channels is exactly what the router wants, and whether those details also include things that are human-readable or not is kinda beside the point.